### ISSUE TITLE
dynamically link against libev if available

### ISSUE THREAD

**Issue by filippog** [armon/statsite#110]
hi,
I'm attempting to package statsite for debian and since libev is shipped as a shared library there I think it'd make sense to dynamically link against that if found by scons when building. I'll followup with a PR too  to get a better idea.

**Comment by filippog** [comment]
I'm getting some warnings when building with external libev tho

```
gcc -o src/networking.o -c -g -std=c99 -D_GNU_SOURCE -O3 -pthread -Ideps/inih/ -Isrc/ src/networking.c
In file included from src/networking.c:39:0:
/usr/include/ev.h:526:26: warning: 'ev_supported_backends' used but never defined
 EV_API_DECL unsigned int ev_supported_backends (void) EV_THROW;
                          ^
/usr/include/ev.h:527:26: warning: 'ev_recommended_backends' used but never defined
 EV_API_DECL unsigned int ev_recommended_backends (void) EV_THROW;
                          ^
/usr/include/ev.h:578:17: warning: 'ev_default_loop' used but never defined
 EV_API_DECL int ev_default_loop (unsigned int flags EV_CPP (= 0)) EV_THROW; /* returns true when successful */
                 ^
/usr/include/ev.h:598:18: warning: 'ev_loop_destroy' used but never defined
 EV_API_DECL void ev_loop_destroy (EV_P);
                  ^
/usr/include/ev.h:633:18: warning: 'ev_run' used but never defined
 EV_API_DECL int  ev_run (EV_P_ int flags EV_CPP (= 0));
                  ^
In file included from src/networking.c:39:0:
/usr/include/ev.h:659:18: warning: 'ev_set_userdata' used but never defined
 EV_API_DECL void ev_set_userdata (EV_P_ void *data) EV_THROW;
                  ^
/usr/include/ev.h:660:19: warning: 'ev_userdata' used but never defined
 EV_API_DECL void *ev_userdata (EV_P) EV_THROW;
                   ^
/usr/include/ev.h:747:18: warning: 'ev_io_start' used but never defined
 EV_API_DECL void ev_io_start       (EV_P_ ev_io *w) EV_THROW;
                  ^
/usr/include/ev.h:748:18: warning: 'ev_io_stop' used but never defined
 EV_API_DECL void ev_io_stop        (EV_P_ ev_io *w) EV_THROW;
                  ^
/usr/include/ev.h:750:18: warning: 'ev_timer_start' used but never defined
 EV_API_DECL void ev_timer_start    (EV_P_ ev_timer *w) EV_THROW;
                  ^
/usr/include/ev.h:751:18: warning: 'ev_timer_stop' used but never defined
 EV_API_DECL void ev_timer_stop     (EV_P_ ev_timer *w) EV_THROW;
                  ^
```

**Comment by filippog** [comment]
fixed in latest updates to PR #111

**Comment by armon** [comment]
Please see my comment on the PR. This is not a path I want to go down.
